### PR TITLE
Add sort algorithms and clean code

### PR DESCRIPTION
**Pull request by yvan-sraka**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#[test]
+fn it_works() {
+    fuhfeuhfe
```
Comment (conmarap) [inline src/lib.rs]: Lets not merge this.

File: Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[package]
+name = "algorithms"
+version = "0.1.0"
+authors = ["Yvan Sraka <yvan.sraka@gmail.com>"]
```
Comment (conmarap) [inline Cargo.toml]: We should add my name to the list here.

File: src/sorts/sleep_sort.rs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+use std::thread;
+
+fn sleepsort<I: Iterator<Item=u32>>(nums: I) {
```
Comment (conmarap) [inline src/sorts/sleep_sort.rs]: Maybe this should be documented. What does it do?

**Comment by yvan-sraka** [comment]
I fix my messy Pull-Request following your comments..

**Comment by conmarap** [comment]
Thank you sir!

**Pull request merged**